crates/pipeline-types/src/config.rs

```diff
@@ -167,3 +172,31 @@ pub struct FormatConfig {
     #[schema(value_type = Object)]
     pub config: YamlValue,
 }
+
+#[derive(Debug, Clone, Eq, PartialEq, Default, Serialize, Deserialize, ToSchema)]
+pub struct ResourceConfig {
+    /// The minimum number of CPU cores to reserve
+    /// for an instance of this pipeline
+    #[serde(default)]
+    pub cpu_min: Option<u16>,
+
+    /// The maximum number of CPU cores to reserve
+    /// for an instance of this pipeline
+    #[serde(default)]
+    pub cpu_max: Option<u16>,
+
+    /// The minimum memory in Megabytes to reserve
```

Comment:
I'd change all these to just bytes.. it's easier to remember that all units are always in unit types rather than having to remember which one is bytes, megabytes or gigabytes

Reply:
I want the smallest unit of storage and memory to be 1MB, hence the choice here.

I'll make the names explicit instead: `memory_mb_min`, `memory_mb_max`.

Reply:
I'd assume we mean Mebibyte as opposed to Megabyte here?

Reply:
no we mean mega (1000^2) not mebi (1024^2).

Reply:
Didn't want to use mebi? seems like the saner unit here :P